### DISCUSSION TITLE
Possibility to select images with less installed tools

### DISCUSSION THREAD

**Discussion by frejonb** [actions/virtual-environments#9569] [Installed software]
Hi all,

In the standard GitHub-hosted runners, we have hit the storage limit when trying to perform heavy builds inside the action. This is typically alleviated by freeing up space via this action https://github.com/marketplace/actions/free-disk-space-ubuntu. However, the more tools we need to delete, the longer the job takes to complete (and therefore a higher cost). Ideally, I would like to select a base runner image that only has the tools we need. However, this is not currently possible for GitHub-hosted runners. 

I'm curious what is other people's experience on this. And what would be a more ideal solution around this problem. I can imagine the more tools people require in the default runner, the less space we'll have for doing actual work.

**Comment by mikhailkoliada** [comment]
Best solution really depends in this case :) I know as we not yet provide small images lots of people tend to build their own images based on the runner-images repo (just take the repo, and cut the installers they do not need).

**Reply by frejonb** [reply]
That makes sense, but would it be possible to run such image in a GitHub-hosted runner? My experience with self hosted runners is not great, especially when trying to maintain enough hot instances all the time for parallel jobs. The main problem is that, as time goes on, jobs that used to complete in our repo now start throwing out-of-storage errors with no changes on our side; just as the size of the runner image increases and the storage capacity of the runner remains the same.

**Reply by crohr** [reply]
If you'd like to reconsider self-hosted runners, you could have a look at [RunsOn](https://runs-on.com), which is very easy to install and operate and allows you to select a custom disk size.
